### PR TITLE
Update default roles template

### PR DESCRIPTION
**Pull request by stevenGravy**
Set v5 as the default and change the default resource rules to more limited RBAC then everything.  Addresses https://github.com/gravitational/teleport/issues/10807

### CODE REVIEW

**Comment by stevenGravy** [comment]
This should be targeted for v9 since this is part of the Testing problems

File: packages/teleport/src/Roles/templates/role.yaml [hunk 1]
```diff
@@ -6,20 +6,55 @@ spec:
   # This example defines an administrative role. It maps to Kubernetes "admin"
   # group and allows SSH to every node.
   allow:
+    # List of Kubernetes cluster users can access the k8s API
+    kubernetes_labels:
+      '*': '*'
     # This role is mapped to Kubernetes 'admin' group.
     kubernetes_groups: [admin]
-
+
```
Comment (zmb3) [inline packages/teleport/src/Roles/templates/role.yaml:14]: Extra whitespace here.

File: packages/teleport/src/Roles/templates/role.yaml [hunk 2]
```diff
@@ -6,20 +6,55 @@ spec:
...
+    db_users:
+    - dba-admin
+    
+    # List of windows desktop access labels that users can open desktop sessions to
+    windows_desktop_labels:
+      '*': '*'
+    # Windows logins a user is allowed to use for desktop sessions.
+    windows_desktop_logins:
+    - '{{internal.windows_logins}}'
+    - Administrator
```
Comment (zmb3) [inline packages/teleport/src/Roles/templates/role.yaml:42]: If we didn't make `root` a default Unix login, then we probably shouldn't make `Administrator` a default Windows login.

Comment (stevenGravy) [reply packages/teleport/src/Roles/templates/role.yaml]: I changed to use developer for that and other examples

File: packages/teleport/src/Roles/templates/role.yaml [hunk 3]
```diff
@@ -3,15 +3,15 @@ metadata:
...
+  # with typical developer credentials.
   allow:
     # List of Kubernetes cluster users can access the k8s API
     kubernetes_labels:
       '*': '*'
-    # This role is mapped to Kubernetes 'admin' group.
-    kubernetes_groups: [admin]
-    
+    # This role is mapped to Kubernetes 'developer' group.
+    kubernetes_groups: [developer]
```
Comment (benarent) [inline packages/teleport/src/Roles/templates/role.yaml:13]:
```suggestion
    kubernetes_groups:
    - '{{internal.kubernetes_groups}}'
    - developer
 ```

File: packages/teleport/src/Roles/templates/role.yaml [hunk 4]
```diff
@@ -3,15 +3,15 @@ metadata:
...
+
```
Comment (benarent) [inline packages/teleport/src/Roles/templates/role.yaml:14]:
```suggestion
    kubernetes_users:
    - '{{internal.kubernetes_users}}'
    - 'dev'
```

File: packages/teleport/src/Roles/templates/role.yaml [hunk 5]
```diff
@@ -31,15 +31,14 @@ spec:
...
     db_users:
-    - dba-admin
+    - developer
     
     # List of windows desktop access labels that users can open desktop sessions to
     windows_desktop_labels:
       '*': '*'
     # Windows logins a user is allowed to use for desktop sessions.
     windows_desktop_logins:
-    - '{{internal.windows_logins}}'
```
Comment (benarent) [inline packages/teleport/src/Roles/templates/role.yaml]: I would keep {{internal}} it's a good time to educe the user on what it is.

Comment (stevenGravy) [reply packages/teleport/src/Roles/templates/role.yaml]: done for all applicable

**Review by zmb3 — APPROVED** [review]

**Review by benarent — APPROVED** [review]

**Comment by benarent** [comment]
Can you also backport to the teleport-9 branch,

**Pull request merged**